.github/workflows/codeql.yml

```
@@ -0,0 +1,155 @@
+        # Deny packages with known vulnerabilities
+        deny-licenses: GPL-2.0, GPL-3.0
+        # Allow certain license types
+        allow-licenses: MIT, Apache-2.0, BSD-2-Clause, BSD-3-Clause, ISC
+        # Comment on PR with summary
+        comment-summary-in-pr: true
+
+  semgrep:
+    name: Semgrep Security Scan
+    runs-on: ubuntu-latest
+    if: github.event_name != 'schedule'
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)